test/widgets/message_list_test.dart

```diff
@@ -300,4 +300,87 @@ void main() {
       debugNetworkImageHttpClientProvider = null;
     });
   });
+
+  group('_UnreadMarker animations', () {
+    Animation getAnimation(WidgetTester tester, int messageId) {
+      final widget = tester.widget<FadeTransition>(find.descendant(
+        of: find.byKey(ValueKey(messageId)),
+        matching: find.byType(FadeTransition)));
+      return widget.opacity;
```

Comment:
This feels kind of brittle. But I don't have a better suggestion off the top of my head, and I'll be OK with merging it. If we play with it we might be able to find something better.

Reply:
I agree, it's unfortunate that the `AnimatedOpacity` widget that is being used in the tree doesn't give us access to the live values and if flutter internally changes how the widget is implemented this would break. Also, if there are further animations added into the `MessageItem` widget (perhaps we have animations on the reactions, or another UI element in each message) this will likely find the wrong one. I've tried playing with it but haven't found anything that isn't equally brittle.